pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -385,3 +386,21 @@ func getSpotVMOptions(spotVMOptions *machinev1.SpotVMOptions) (compute.VirtualMa
 	// https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/spot-instances.md#eviction-policies
 	return compute.VirtualMachinePriorityTypesSpot, compute.VirtualMachineEvictionPolicyTypesDeallocate, billingProfile, nil
 }
+
+func generateImagePlan(image machinev1.Image) *compute.Plan {
+	// If it's not a third-party image or from the shared gallery,
+	// we don't need a purchase plan.
+	if !image.ThirdPartyImage && image.SharedGallery == nil {
+		return nil
+	}
+
+	if image.Publisher == "" || image.SKU == "" || image.Offer == "" {
+		return nil
```

Comment:
Do we care about the version? IIRC the upstream implementation cared about versions and put those into the plan? I could be mistaken though

Reply:
We don't care about version in the plan. 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/d2b4e8ed2313adc270a0911d9d0c64e344f34cc4/azure/services/virtualmachines/virtualmachines.go#L278